cmd/dmverity-vhd/main.go

```diff
@@ -98,7 +100,32 @@ func fetchImageLayers(ctx *cli.Context) (layers []v1.Layer, err error) {
 
 	// by default, using remote as source
 	var img v1.Image
-	if tarballPath != "" {
+	if tarballPath != "" || dockerDaemon {
+		if dockerDaemon {
+			ctx := context.Background()
+			cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+			if err != nil {
+				return nil, fmt.Errorf("failed to instanciate docker client: %w", err)
+			}
+
+			imageTarReader, err := cli.ImageSave(ctx, []string{image})
+			if err != nil {
+				return nil, fmt.Errorf("failed to load image tar: %w", err)
+			}
+			defer imageTarReader.Close()
+
+			tarFile, err := os.CreateTemp("", "image-tar")
+			if err != nil {
+				return nil, fmt.Errorf("failed to create tar file: %w", err)
+			}
+			defer tarFile.Close()
```

Comment:
It looks like `defer os.Remove(tarFile.Name())` is also needed to clean up

https://pkg.go.dev/os#CreateTemp

Reply:
I can't quite do that otherwise the tar file is cleaned up before it's read for layer hashing.
But after hashing we can cleanup any temporary file matching the pattern we use to create the tar.

Done in https://github.com/microsoft/hcsshim/pull/2086/commits/23521400b67d091318e3d26528e1be70a4c3d3e0 and https://github.com/microsoft/hcsshim/pull/2086/commits/d79678f73e8e59cca1fad92bab4c903fed27265f